mesonbuild/scripts/env2mfile.py

```diff
@@ -48,20 +48,34 @@ def add_arguments(parser: 'argparse.ArgumentParser') -> None:
                         help='Define endianness for cross compilation.')
 
 class MachineInfo:
-    def __init__(self) -> None:
-        self.compilers: T.Dict[str, T.List[str]] = {}
-        self.binaries: T.Dict[str, T.List[str]] = {}
-        self.properties: T.Dict[str, T.Union[str, T.List[str]]] = {}
-        self.compile_args: T.Dict[str, T.List[str]] = {}
-        self.link_args: T.Dict[str, T.List[str]] = {}
-        self.cmake: T.Dict[str, T.Union[str, T.List[str]]] = {}
-
-        self.system: T.Optional[str] = None
-        self.subsystem: T.Optional[str] = None
-        self.kernel: T.Optional[str] = None
-        self.cpu: T.Optional[str] = None
-        self.cpu_family: T.Optional[str] = None
-        self.endian: T.Optional[str] = None
+    def __init__(
+        self,
+        compilers: T.Dict[str, T.List[str]] = {},
```

Comment:
pylint doesn't like this, but I'm using `dict()` to explicitly copy it, which avoids the pitfalls of mutable default arguments. I'll see whether there's a common solution to this elsewhere in the codebase, but in the meantime I'd appreciate review on the rest.